### DISCUSSION TITLE
Handling Activation Redirection in Layout with Next.js 13 App Router Without Page Render Execution

### DISCUSSION THREAD

**Discussion by haloworld007** [vercel/next.js#58205] [App Router]
### Next.js version
13.5.7 app router

### Summary
I am working with Next.js 13's new App Router, and I'm trying to implement a global check in my layout to determine if a user is activated. If the user is not activated, I want to redirect them to an activation page. The problem is that even after a redirect is initiated, the page's rendering logic is still executed.

### Details
- I need the check to happen globally, at the layout level, and be executed only once per session or upon state change.
- The redirection works, but the rendering of the originally requested page still occurs, which is not the intended behavior.

### Question
Is there a built-in mechanism or recommended pattern in Next.js for halting the execution of a page's rendering logic if a redirect is performed at the layout level?

### Additional Context
The aim is to ensure that pages that require user activation do not render until the activation check is passed, optimizing both security and performance.

Any suggestions or guidance on how to address this within the Next.js framework would be greatly appreciated.

**Comment by bacqueyrisses** [comment]
Same question. I have to do the redirection on the page level for it to work and not render the children.

**Reply by TheDeafOne** [reply]
Ditto, this would be really useful to know

**Comment by RoSHan0008** [comment]
you can use loading page it will resolved your issue

**Comment by saurabhsri108** [comment]
I think @haloworld007, this can be raised as an issue as this persists even in 14.2.5 version.

@leerob is it a pattern or feature?
